acorn/src/expression.js

```diff
@@ -262,7 +262,8 @@ pp.parseExprSubscripts = function(refDestructuringErrors) {
 
 pp.parseSubscripts = function(base, startPos, startLoc, noCalls) {
   let maybeAsyncArrow = this.options.ecmaVersion >= 8 && base.type === "Identifier" && base.name === "async" &&
-      this.lastTokEnd === base.end && !this.canInsertSemicolon() && this.input.slice(base.start, base.end) === "async"
+      this.lastTokEnd === base.end && !this.canInsertSemicolon() && base.end - base.start === 5 &&
```

Comment:
`base.name === "async" && base.end - base.start === 5` should implies that `this.input.slice(base.start, base.end) === "async"`, given any escaped sequence rendering strictly longer base length.